internal/authentication/oidc/clientcredentials.go

```diff
@@ -34,6 +34,17 @@ const (
 	oidcScopeOpenID = "openid"
 )
 
+// ClientCredentialsMetadata is the metadata fields which can be used by a
+// component to configure an OIDC client_credentials token source.
+type ClientCredentialsMetadata struct {
+	TokenCAPEM   string   `mapstructure:"oidcTokenCAPEM"`
+	TokenURL     string   `mapstructure:"oidcTokenURL"`
+	ClientID     string   `mapstructure:"oidcClientID"`
+	ClientSecret string   `mapstructure:"oidcClientSecret"`
+	Audiences    []string `mapstructure:"oidcAudiences"`
+	Scopes       []string `mapstructure:"oidcScopes"`
```

Comment:
These are OAuth2 credentials not OIDC (In fact, it's called "Client Credentials" which is an OAuth2 flow :) ). Since these names are going to be used for OAuth2 (without OIDC) they should be generic.

Unsure if `oauth` or `oauth2` is better.

```suggestion
	TokenCAPEM   string   `mapstructure:"oauth2TokenCAPEM"`
	TokenURL     string   `mapstructure:"oauthT2okenURL"`
	ClientID     string   `mapstructure:"oauth2ClientID"`
	ClientSecret string   `mapstructure:"oauth2ClientSecret"`
	Audiences    []string `mapstructure:"oauth2Audiences"`
	Scopes       []string `mapstructure:"oauth2Scopes"`
```

Reply:
I've gone ahead and renamed it all to be OAuth2 and removed all OIDC compliance checks. It is now simply a generic client_credentials OAuth2 authenticator. The docs will need to be written so that the user is aware that they are responsible for making sure the authenticater is OIDC compliant. The certification tests have been kept basically the same since pulsar is expecting an OIDC provider be available and OIDC client tokens.

IMO, I think it is a mistake to not make this an ODIC authenticator but am keen to unblock users who want to use OIDC authentication in pulsar.